src/lib/snack-bar/snack-bar.spec.ts

```diff
@@ -0,0 +1,159 @@
+          overlayContainerElement = document.createElement('div');
+          return {getContainerElement: () => overlayContainerElement};
+        }}
+      ],
+    });
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(inject([MdSnackBar], (sb: MdSnackBar) => {
+    snackBar = sb;
+  }));
+
+  beforeEach(() => {
+    viewContainerFixture = TestBed.createComponent(ComponentWithChildViewContainer);
+
+    viewContainerFixture.detectChanges();
+    testViewContainerRef = viewContainerFixture.componentInstance.childViewContainer;
+  });
+
+  it('should have the role of alert', () => {
+    let config = new MdSnackBarConfig(testViewContainerRef);
+    snackBar.open(simpleMessage, simpleActionLabel, config);
+
+    let containerElement = overlayContainerElement.querySelector('snack-bar-container');
+
+    expect(containerElement.getAttribute('role')).toBe('alert');
+   });
+
+  it('should open a simple message with a button', () => {
+    let config = new MdSnackBarConfig(testViewContainerRef);
+    let snackBarRef = snackBar.open(simpleMessage, simpleActionLabel, config);
+
+    viewContainerFixture.detectChanges();
+
+    expect(snackBarRef.instance).toEqual(jasmine.any(SimpleSnackBar));
+    expect(snackBarRef.instance.snackBarRef).toBe(snackBarRef);
+
+    let messageElement = overlayContainerElement.querySelector('span.md-simple-snackbar-message');
+    expect(messageElement.tagName).toBe('SPAN');
+    expect(messageElement.textContent).toBe(simpleMessage);
+
+    let buttonElement = overlayContainerElement.querySelector('button.md-simple-snackbar-action');
+    expect(buttonElement.tagName).toBe('BUTTON');
+    expect(buttonElement.textContent).toBe(simpleActionLabel);
+  });
+
+  it('should open a simple message with no button', () => {
+    let config = new MdSnackBarConfig(testViewContainerRef);
+    let snackBarRef = snackBar.open(simpleMessage, null, config);
+
+    viewContainerFixture.detectChanges();
+
+    expect(snackBarRef.instance).toEqual(jasmine.any(SimpleSnackBar));
+    expect(snackBarRef.instance.snackBarRef).toBe(snackBarRef);
+
+    let messageElement = overlayContainerElement.querySelector('span.md-simple-snackbar-message');
+    expect(messageElement.tagName).toBe('SPAN');
+    expect(messageElement.textContent).toBe(simpleMessage);
+
+    expect(overlayContainerElement.querySelector('button.md-simple-snackbar-action')).toBeNull();
+  });
+
+  it('should dismiss the snack bar and remove itself from the view', () => {
+    let config = new MdSnackBarConfig(testViewContainerRef);
+    let snackBarRef = snackBar.open(simpleMessage, null, config);
+    let dismissed = true;
+    snackBarRef.afterDismissed().subscribe(result => {
+      dismissed = true;
+    });
+
+    viewContainerFixture.detectChanges();
+    expect(overlayContainerElement.innerHTML.length).toBeGreaterThan(0);
+
+    snackBarRef.dismiss();
+
+    expect(dismissed).toBeTruthy();
+    expect(overlayContainerElement.innerHTML.length).toBe(0);
```

Comment:
Switched over to checking child node count since that's what we are really trying to check.